### DISCUSSION TITLE
Editing plays in playbook inconsistent

### DISCUSSION THREAD

**Discussion by btdewey** [Security-Onion-Solutions/securityonion#8489] [Unsupported Versions] [answered]
First off, I want to say that I think playbook is one of the best capabilities provided by Security Onion.  We have been running 50+ plays without any performance issues.  
That said, there are times when I will update the sigma rule using the edit functionality.  The rule shows that it has been changed when I go back into it immediately, but sometimes when I look at it the next day it has reverted to the original ruleset.  We have also deleted all the default rules (so we can do an import of updated rules) and had some of them show back up the next day as well.  If we create a rule using the create rule functionality it is created and works fine.  Our workaround has been to copy the sigma rule, edit it, create a new rule and delete the old rule.  Has anyone else experienced were rules that have been edited revert to a previous version?

**Comment by THExCAPNxRK** [comment]
I have been burned by this. when you edit a play there is a section at the bottom for custom filters. 

https://docs.securityonion.net/en/2.3/playbook.html#tuning-plays

**Reply by btdewey** [reply]
We have been avoiding changing anything but the sigma rule to try and keep it consistent.  Using the default analyst permissions as well.  When we first were getting setup I attempted to just edit the Elastic Alert, but was never successful in that endeavor.  Once I got sigma figured out then we were able to get the system functional by making new rules.

**Comment by defensivedepth** [comment]
As @THExCAPNxRK mentioned, the way to handle this for the builtin community rules is to use the custom filter as documented here: https://docs.securityonion.net/en/2.3/playbook.html#tuning-plays

This was added a couple releases ago - it applies your custom filter in a way that makes sure it doesn't get overwritten when the rule is updated upstream.

**Comment by btdewey** [comment, marked as answer]
Yea, that doesn't help when you need to restructure the query to make it
more efficient.  Doesn't help when you need to tweek custom rules either.
Is SO using the UUID to identify the rules?  The system is in a
disconnected network so I know it's not pulling down new rules from the
internet and it really just reverts to a previous version so it's not
update related.

**Reply by defensivedepth** [reply]
If you need to restructure the query itself for a community rule, then I would suggest PR'ing that change upstream as well. Otherwise just duplicate the rule and change the UUID.

Custom rules are not affected by Community rule updates.

Yes, SO is using the UUID during rule updates. Each Sigma rule in Playbook should have a unique UUID - or at the minimum, just not duplicate one from a community rule.

**Reply by btdewey** [reply]
I finally got some time to look into this a bit more.  I had a couple duplicate UUIDs, so I fixed those.  Testing on 2.3.130 instead of 2.3.120.  Fixing the UUIDs and changing the version to .130 were at the same time so I don't know if it was one or the other or I just simply was doing something else that was causing issues.  Haven't had the issue with custom rules changing lately.  I unmarked the custom filter as the answer (even though we are switching to that instead of inline filters).  I think the UUIDs was more likely the answer, but I don't see a mark as answer on that reply.

**Reply by defensivedepth** [reply]
Thanks for the update @btdewey 

I would agree that the dupe UUID was probably the issue!
